### DISCUSSION TITLE
EtherDFS linked article steers away from using EtherDFS

### DISCUSSION THREAD

**Discussion by jamescadd** [retronas/discussions#74] [Q&A] [answered]
In reading through the links from the wiki I found this article: https://pierre32.medium.com/etherdfs-and-retronas-the-problem-and-the-fix-390eb9584c94

It's pretty recent so I expect its accurate.  The author describes issues transferring files with EtherDFS and later resolved them by switching to ProFTPd.

I am curious if anyone else has had issues with EtherDFS as I'm not sure which direction to go based on that article.  A mapped DOS drive sounds convenient, but if it's flaky or falls down on transferring large directories it's probably not worth the effort for me.

**Comment by danmons** [comment, marked as answer]
The issue was EtherDFS's dislike of filenames that weren't in 8.3 format. This can be resolved a few ways:

1) Ensuring all filenames are in 8.3 format. Sounds trite, but there's no original DOS software that wasn't. This tends only to happen with modern rebundling of old software (for example, GOG bundles). The same problem doesn't exist with things like the eXoDOS bundles. 

2) Don't use the "vfat" file system mount type in Linux (mounts FAT filesystems with modern long filename support), but rather use the "msdos" to force-truncate long file names. 

EtherDFS is perfectly stable for regular use as a mounted drive letter. The issue here are modern long filenames for an old OS that doesn't support them. Good housekeeping of era-appropriate filenames is a critical part of retro computing. 

If it's not already, I'll try to add some notes to this effect into the wiki.

**Comment by jamescadd** [comment]
Thanks Dan!  This was not in the wiki as far as I saw, and it's not mentioned in the articles linked either.  I do see that there was some discussion on 8.3 filenames on Vogons so there may be some useful tidbits there too.  Appreciate the response.

**Comment by jamescadd** [comment]
@danmons A couple of ideas for when you add docs to EtherDFS based on my experience so far trying to get up and running:
- In the filesystems part of the wiki, you could make a note not to use VFAT as you indicated here.  On the wiki it currently reads like VFAT is just what Linux calls FAT filesystems.  I was about to format a drive as VFAT in Cockpit in an attempt to get started, then re-read this discussion and realized it wouldn't work.
- You have many good setup ideas on the Vogons thread for scenarios you plan to cover in videos!  Choosing whatever is the easiest and documenting it as a quick way to get people started may help them get off the ground.  Right now the guidance isn't complete enough to get me up and running with my minimal Linux experience.
- You mention mounting a drive as msdos - that's new to me and would be great to cover in the docs.  You've historically done an excellent job of explaining Linux basics like this in your videos, which I have appreciated.
- For those of us running on "not a Pi" it would be nice to have a way to do get going that doesn't require an attached USB drive.  Bonus if its something that can be located on a NAS (my current RetroNAS install on ProxMox is backed by TrueNAS for the filesystem).
- Bonus: Have EtherDFS install just set this stuff up as a virtual disk or similar when its installed.  I would rather take the defaults, hop on the client and install those bits and be up and running than think about config.  Personal preference I suppose :)
